pyccl/halos/profiles/profile_base.py

```diff
@@ -47,35 +47,31 @@ def __init__(self):
                             "_real or _fourier implementation.")
         self.precision_fftlog = FFTLogParams()
 
-    @property
-    @abstractmethod
-    def normprof(self) -> bool:
-        """Normalize the profile in auto- and cross-correlations by
-        :math:`I^0_1(k\\rightarrow 0, a|u)`
-        (see :meth:`~pyccl.halos.halo_model.HMCalculator.I_0_1`).
-        """
-
-    # TODO: CCLv3 - Rename & allocate _normprof_bool to the subclasses.
+    def get_normalization(self, cosmo, a, hmc):
+        """Profiles may be normalised by an overall function of redshift
+        (or scale factor). This function may be cosmology dependent and
```

Comment:
Yet another reason to make everything kw-only, because not everything depends on `cosmo`, someone might want to implement it like `def get_normalization(self, *, a, hmc, **kwargs)`.

Reply:
Again, we shouldn't allow that. This is meant to be a more restrictive version of `fourier` or `real`, for which we don't allow additional dependence.

Reply:
yep - what you proposed in your other comment works, so I'm resolving this!